Extensions/xkit_patches.js

```diff
@@ -129,6 +129,52 @@ XKit.extensions.xkit_patches = new Object({
 	patches: {
 		"7.9.0": function() {
 
+			XKit.svc = {
+				blog: {
+					followed_by: data => new Promise((resolve, reject) => {
+						XKit.tools.Nx_XHR({
+							method: "GET",
+							url: "https://www.tumblr.com/svc/blog/followed_by?" + $.param(data),
+							onload: resolve,
+							onerror: reject
+						});
+					})
+				},
+
+				conversations: {
+					participant_info: data => new Promise((resolve, reject) => {
+						XKit.tools.Nx_XHR({
+							method: "GET",
+							url: "https://www.tumblr.com/svc/conversations/participant_info?" + $.param(data),
+							onload: resolve,
+							onerror: reject
+						});
+					})
+				}
+			};
+
+			/**
+			 * Determines whether a user is following the given blog.
+			 * The logged-in user must be a member of the given blog to determine this.
+			 * @param {String} username
+			 * @param {String} blog
+			 * @return {Promise<Boolean>}
+			 */
+			XKit.interface.is_following = function(username, blog) {
+				return new Promise(resolve => {
+					XKit.svc.conversations.participant_info({
```

Comment:
is this guaranteed to work for, e.g., people who have messaging turned off?

Reply:
no but thats what the first `catch` is for!

Reply:
So the new behavior is that it will work all the time for mainblogs, but only work partially for sideblogs? Isn't that worse then not working at all for sideblogs?

Reply:
you can't turn messaging off without becoming a group blog though? and usually group blogs make it clear that they're group blogs